contracts/Obligation.sol

```diff
@@ -157,19 +163,24 @@ contract Obligation is IObligation, IERC20, European, Ownable, WriterRegistry {
     }
 
     /**
-     * @notice Mints obligation tokens
-     * @dev Can only be called by option contract before expiry
-     * @param account Address to credit
+     * @notice Mints obligation tokens to an `account` transfers the respective option
+     * tokens to a `pool` - e.g. Uniswap. To prevent misappropriation of funds we expect
+     * this function to be called atomically after depositing in the treasury. The `OptionLib`
+     * contract should provide helpers to facilitate this.
+     * @dev Once the expiry date has lapsed this function is no longer valid.
+     * @param account Origin address
+     * @param pool Destination address (i.e. uniswap pool)
      * @param amount Total credit
      * @param btcHash Bitcoin hash
      * @param format Bitcoin script format
      **/
     function mint(
         address account,
+        address pool,
         uint256 amount,
         bytes20 btcHash,
         Bitcoin.Script format
-    ) external override notExpired onlyOwner {
```

Comment:
This should remain to be onlyOwner or only be callable internally. Otherwise, anyone can generate obligation tokens that can possibly be sold on Uniswap.

Reply:
This is the entrypoint to obligation / option writing which is called by the `OptionLib` for instance. Note that we require the `account` to have sufficient balance in the treasury in order for this call to be valid.

Reply:
You can also see that this modifier has been moved to the equivalent method on the `Option` contract since we have reversed the ownership roles.

Reply:
The function can currently be called directly from any account. If that is done the treasury will try to lock more collateral. If that succeeds because the user has deposited, we generate obligation tokens without also generating option tokens. That is not the desired behaviour.

I agree that you have to kind of force this and on a low level do this, but still, we should not allow it either way. We could

Reply:
So we can only lock 'unlocked' collateral, meaning the user has to atomically supply unlocked collateral then mint (as in Uniswap) - see `lockAndWrite` in `OptionLib.sol`. Once the treasury atomically locks this collateral, preventing additional claims, we then mint the option tokens. Unless we whitelist addresses that can call `mint` here it needs to be external to allow for different 'routers' to do these kinds of atomic operations.

Reply:
But you could atomically supply, unlock and mint obligation tokens without generating option tokens?

Reply:
We call the `mint` function on the option contract as well: https://github.com/interlay/xopts/pull/28/files#diff-c1d6f4d4cca0713bf9049a9a5b64dd8eR197

Reply:
Ah sorry, my bad! Ok, then this is good to go!